Makefile

```diff
@@ -19,24 +29,19 @@ override DEPLOY_ARGS += --service_discovery
 
 # Targets to make
 
-# Explicitly depend on the binary, since if it doesn't exist Shipyard won't find it
-package/.image.lighthouse-agent: bin/lighthouse-agent
-
-package/.image.lighthouse-coredns: bin/lighthouse-coredns
-
-build: $(BINARIES)
+build: $(ARCH_BINARIES)
 
-bin/lighthouse-agent: vendor/modules.txt $(shell find pkg/agent)
-	${SCRIPTS_DIR}/compile.sh $@ pkg/agent/main.go $(BUILD_ARGS)
+bin/%/lighthouse-agent: vendor/modules.txt $(shell find pkg/agent)
+	GOARCH=$(call dockertogoarch,$(patsubst bin/linux/%/,%,$(dir $@))) ${SCRIPTS_DIR}/compile.sh $@ pkg/agent/main.go $(BUILD_ARGS)
 
-bin/lighthouse-coredns: vendor/modules.txt $(shell find pkg/coredns)
-	${SCRIPTS_DIR}/compile.sh $@ pkg/coredns/main.go $(BUILD_ARGS)
+bin/%/lighthouse-coredns: vendor/modules.txt $(shell find pkg/coredns)
+	GOARCH=$(call dockertogoarch,$(patsubst bin/linux/%/,%,$(dir $@))) ${SCRIPTS_DIR}/compile.sh $@ pkg/coredns/main.go $(BUILD_ARGS)
 
 e2e: vendor/modules.txt
 
 licensecheck: BUILD_ARGS=--noupx
-licensecheck: $(BINARIES) bin/lichen
-	bin/lichen -c .lichen.yaml $(BINARIES)
+licensecheck: $(ARCH_BINARIES) bin/lichen
+	bin/lichen -c .lichen.yaml $(ARCH_BINARIES)
```

Comment:
I wonder if it's really necessary to do it for all the binaries, and not just one arch (shouldnt they have similar license regardless of platform?)

Reply:
Some platforms result in different dependencies (_e.g._ Windows).